src/components/forms/Dropdown/Dropdown.tsx

```diff
@@ -1,52 +1,11 @@
 /*
-  @deprecated Make updates to the Select component instead
-  @todo Remove this component
+  TODO: Remove this component
 */
 
-import React from 'react'
 import { Select } from '../Select/Select'
 import { withDeprecationWarning } from '../../hoc/withDeprecationWarning'
-import { ValidationStatus } from '../../../types/validationStatus'
 
-type DropdownProps = {
-  id: string
-  name: string
-  className?: string
-  children: React.ReactNode
-  validationStatus?: ValidationStatus
-  inputRef?:
-    | string
-    | ((instance: HTMLSelectElement | null) => void)
-    | React.RefObject<HTMLSelectElement>
-    | null
-    | undefined
-}
-
-export const Dropdown = ({
-  id,
-  name,
-  className,
-  inputRef,
-  children,
-  validationStatus,
-  ...inputProps
-}: DropdownProps & JSX.IntrinsicElements['select']): React.ReactElement => {
-  const deprecationWarningMessage = 'Dropdown is deprecated and will be removed in the future. Please use the Select component instead.'
-  const DeprecatedDropdown = withDeprecationWarning(
-    Select,
-    deprecationWarningMessage
-  )
-  return (
-    <DeprecatedDropdown
-      validationStatus={validationStatus}
-      className={className}
-      id={id}
-      name={name}
-      ref={inputRef}
-      {...inputProps}>
-      {children}
-    </DeprecatedDropdown>
-  )
-}
-
-export default Dropdown
+export const Dropdown = withDeprecationWarning(
+  Select,
+  'Dropdown is deprecated and will be removed in the future. Please use the Select component instead.'
+)
```

Comment:
@werdnanoslen this was the main thing I wanted to suggest. `withDeprecationWarning` allows you to wrap any component, and it just returns a _new_ component with the name you give it, which renders with a `useEffect` that calls `deprecationWarning` on initial render, and also renders the component you pass in.

So no need to define and pass around props, it basically inherits all of that from Select. `Dropdown` is now just `Select` with the old name, that also renders a warning.

🪄 Magic 🪄

Reply:
ahh got it, haven't used a pattern like this (intentionally) before, but that makes sense!